client/my-sites/checkout/modal/index.tsx

```diff
@@ -60,7 +61,11 @@ const CheckoutModal: FunctionComponent< Props > = ( {
 
 	const handleRequestClose = () => {
 		onClose?.();
-		navigate( cancelTo );
+		if ( isRelativeUrl( cancelTo ) ) {
+			navigate( cancelTo );
+		} else {
+			navigate( previousRouteWithArgs );
+		}
```

Comment:
I don't think this code is user-triggerable. It could be run in some situations but I'm not sure what those are (if even are any)